src/ng/compile.js

```diff
@@ -969,6 +969,17 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
     return require;
   }
 
+  function getDirectiveRestrict(restrict, name) {
+    if (restrict && !(isString(restrict) && /[EACM]/.test(restrict))) {
```

Comment:
I've updated this to convert all falsey values to the default like previously.

Cases which previously logged errors per-compilation (of an attribute/element with the same name), but otherwise did nothing:
* non-string truthy values

Cases which previously silently did nothing:
* arrays without single-character E/A/C/M
* strings with only invalid characters

Cases that previously worked unintentionally:
* arrays with single-character entries of E/A/C/M (such as `['E', 'A']` but not `['EA']`)

Now all of these cases get logged as errors the first time an attribute/element of that name gets compiled.